go/internal/a2a/recording_manager.go

```diff
@@ -0,0 +1,200 @@
+package a2a
+
+import (
+	"context"
+
+	"github.com/kagent-dev/kagent/go/internal/database"
+	"github.com/kagent-dev/kagent/go/internal/utils"
+	"github.com/kagent-dev/kagent/go/pkg/auth"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"trpc.group/trpc-go/trpc-a2a-go/client"
+	"trpc.group/trpc-go/trpc-a2a-go/protocol"
+	"trpc.group/trpc-go/trpc-a2a-go/taskmanager"
+)
+
+// RecordingManager is similar to the PassthroughManager, but it also stores the task and session information in the database.
+// It is used for remote agents, which do not have a connection to kagent, so we use this middleware to handle that.
+// Note: We expect the remote agent to handle its task history + token information which we store and parse from.
+type RecordingManager struct {
+	client   *client.A2AClient
+	dbClient database.Client
+	agentRef string
+}
+
+func NewRecordingManager(client *client.A2AClient, dbClient database.Client, agentRef string) taskmanager.TaskManager {
+	return &RecordingManager{
+		client:   client,
+		dbClient: dbClient,
+		agentRef: agentRef,
+	}
+}
+
+// getUserID extracts user ID from context.
+// similar to _get_user_id in kagent-adk's request_converter.py
+func (m *RecordingManager) getUserID(ctx context.Context, contextID string) string {
+	if session, ok := auth.AuthSessionFrom(ctx); ok {
+		if session.Principal().User.ID != "" {
+			return session.Principal().User.ID
+		}
+	}
+
+	return "A2A_USER_" + contextID
+}
+
+func (m *RecordingManager) OnSendMessage(ctx context.Context, request protocol.SendMessageParams) (*protocol.MessageResult, error) {
+	if request.Message.MessageID == "" {
+		request.Message.MessageID = protocol.GenerateMessageID()
+	}
+	if request.Message.Kind == "" {
+		request.Message.Kind = protocol.KindMessage
+	}
+
+	result, err := m.client.SendMessage(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	if task, ok := result.Result.(*protocol.Task); ok {
+		logger := ctrllog.FromContext(ctx).WithName("recording-manager")
+
+		if m.dbClient != nil {
+			userID := m.getUserID(ctx, task.ContextID)
+
+			// Get agent ID from the agent ref
+			var agentID *string
+			if m.agentRef != "" {
+				agentIDStr := utils.ConvertToPythonIdentifier(m.agentRef)
+				agentID = &agentIDStr
+			}
+
+			session := &database.Session{
+				ID:      task.ContextID,
+				UserID:  userID,
+				AgentID: agentID,
+			}
+
+			// Store a session (fail silently if duplicate)
+			if err := m.dbClient.StoreSession(session); err != nil {
+				logger.Error(err, "Failed to create session", "contextID", task.ContextID, "sessionID", session.ID, "agentID", session.AgentID)
+			}
+
+			// Store Task - always store whatever the remote agent provides
+			if err := m.dbClient.StoreTask(task); err != nil {
+				logger.Error(err, "Failed to store sync task", "taskID", task.ID, "contextID", task.ContextID)
+			}
```

Comment:
Why are we failing silently here? The whole point of this functionality is to record this